logging/api/LoggingTest/LoggingTest.cs

```diff
@@ -0,0 +1,237 @@
+﻿/*
+ * Copyright (c) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using System;
+using Google.Logging.V2;
+using Xunit;
+using System.IO;
+using System.Diagnostics;
+using System.Threading;
+
+namespace GoogleCloudSamples
+{
+    public class BaseTest
+    {
+        private readonly string _projectId;
+
+        public BaseTest()
+        {
+            _projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        }
+
+        public struct ConsoleOutput
+        {
+            public int ExitCode;
+            public string Stdout;
+        };
+
+        /// <summary>Runs LoggingSample.exe with the provided arguments</summary>
+        /// <returns>The console output of this program</returns>
+        public static ConsoleOutput Run(params string[] arguments)
+        {
+            Console.Write("LoggingSample.exe ");
+            Console.WriteLine(string.Join(" ", arguments));
+
+            using (var output = new StringWriter())
+            {
+                LoggingSample loggingSample = new LoggingSample(output);
+                var consoleOutput = new ConsoleOutput()
+                {
+                    ExitCode = loggingSample.Run(arguments),
+                    Stdout = output.ToString()
+                };
+                Console.Write(consoleOutput.Stdout);
+                return consoleOutput;
+            }
+        }
+
+        protected static void AssertSucceeded(ConsoleOutput output)
+        {
+            Assert.True(0 == output.ExitCode,
+                $"Exit code: {output.ExitCode}\n{output.Stdout}");
+        }
+
+        public class LoggingTest : BaseTest
+        {
+            [Fact]
+            public void TestCreateLogEntry()
+            {
+                string logId = "logForTestCreateLogEntry";
+                string message = "Example log entry.";
+                // Try creating a log entry.
+                var created = Run("create-log-entry", logId, message);
+                AssertSucceeded(created);
+                // Pause for 5 seconds before trying to get newly added log entry.
+                Thread.Sleep(5000);
+                // Retrieve the log entry just added, using the logId as a filter.
+                var results = Run("list-log-entries", logId);
+                // Confirm returned log entry contains expected value.
+                Assert.Contains(message, results.Stdout);

```

Comment:
If the assertion fails, then the Run("delete-log") statement below will never run, and the test will leak a log.

This pattern repeats in the logs below.  Adding try .. finally .. to every test would be tedious.  Instead, you could add a list of garbage as a member variable for the class, and then implement Dispose() to delete the garbage.